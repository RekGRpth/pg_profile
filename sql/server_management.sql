/* == Testing server management functions == */
SELECT profile.create_server('srvtest','dbname=postgres host=localhost port=5432', TRUE, NULL, 'Server description 1');
SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
SELECT profile.rename_server('srvtest','srvtestrenamed');
SELECT profile.set_server_connstr('srvtestrenamed','dbname=postgres host=localhost port=5433');
SELECT profile.set_server_description('srvtestrenamed','Server description 2');
SELECT profile.set_server_db_exclude('srvtestrenamed',ARRAY['db1','db2','db3']);
SELECT profile.set_server_max_sample_age('srvtestrenamed',3);
-- settings validation test
SELECT profile.set_server_setting('srvtestrenamed','name_failure','test');
SELECT profile.set_server_setting('srvtestrenamed','collect_vacuum_stats','value_failure');
SELECT profile.set_server_setting('srvtestrenamed','collect_vacuum_stats','on');
SELECT srv_settings::text FROM profile.servers ORDER BY server_id;
SELECT * FROM profile.show_server_settings('srvtestrenamed');
SELECT profile.set_server_setting('srvtestrenamed','collect_vacuum_stats');
SELECT * FROM profile.show_server_settings('srvtestrenamed');
SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
SELECT profile.disable_server('srvtestrenamed');
SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
SELECT profile.enable_server('srvtestrenamed');
SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
SELECT * FROM profile.show_servers() where server_name != 'local';
SELECT * FROM profile.drop_server('srvtestrenamed');
