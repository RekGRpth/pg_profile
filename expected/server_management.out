/* == Testing server management functions == */
SELECT profile.create_server('srvtest','dbname=postgres host=localhost port=5432', TRUE, NULL, 'Server description 1');
 create_server 
---------------
             2
(1 row)

SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
 server_id | server_name |  server_description  | db_exclude | enabled |                 connstr                  | max_sample_age | last_sample_id 
-----------+-------------+----------------------+------------+---------+------------------------------------------+----------------+----------------
         2 | srvtest     | Server description 1 |            | t       | dbname=postgres host=localhost port=5432 |                |              0
(1 row)

SELECT profile.rename_server('srvtest','srvtestrenamed');
 rename_server 
---------------
             1
(1 row)

SELECT profile.set_server_connstr('srvtestrenamed','dbname=postgres host=localhost port=5433');
 set_server_connstr 
--------------------
                  1
(1 row)

SELECT profile.set_server_description('srvtestrenamed','Server description 2');
 set_server_description 
------------------------
                      1
(1 row)

SELECT profile.set_server_db_exclude('srvtestrenamed',ARRAY['db1','db2','db3']);
 set_server_db_exclude 
-----------------------
                     1
(1 row)

SELECT profile.set_server_max_sample_age('srvtestrenamed',3);
 set_server_max_sample_age 
---------------------------
                         1
(1 row)

-- settings validation test
SELECT profile.set_server_setting('srvtestrenamed','name_failure','test');
ERROR:  Unsupported setting
CONTEXT:  PL/pgSQL function set_server_setting(name,text,text) line 41 at RAISE
SELECT profile.set_server_setting('srvtestrenamed','collect_vacuum_stats','value_failure');
ERROR:  Value for collection conditions should be boolean: invalid input syntax for type boolean: "value_failure"
DETAIL:  
CONTEXT:  PL/pgSQL function set_server_setting(name,text,text) line 26 at RAISE
SELECT profile.set_server_setting('srvtestrenamed','collect_vacuum_stats','on');
 set_server_setting 
--------------------
                  1
(1 row)

SELECT srv_settings::text FROM profile.servers ORDER BY server_id;
            srv_settings             
-------------------------------------
 
 {"collect": {"vacuum_stats": true}}
(2 rows)

SELECT * FROM profile.show_server_settings('srvtestrenamed');
  scope  |   setting    | value 
---------+--------------+-------
 collect | vacuum_stats | true
(1 row)

SELECT profile.set_server_setting('srvtestrenamed','collect_vacuum_stats');
 set_server_setting 
--------------------
                  1
(1 row)

SELECT * FROM profile.show_server_settings('srvtestrenamed');
 scope | setting | value 
-------+---------+-------
(0 rows)

SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
 server_id |  server_name   |  server_description  |  db_exclude   | enabled |                 connstr                  | max_sample_age | last_sample_id 
-----------+----------------+----------------------+---------------+---------+------------------------------------------+----------------+----------------
         2 | srvtestrenamed | Server description 2 | {db1,db2,db3} | t       | dbname=postgres host=localhost port=5433 |              3 |              0
(1 row)

SELECT profile.disable_server('srvtestrenamed');
 disable_server 
----------------
              1
(1 row)

SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
 server_id |  server_name   |  server_description  |  db_exclude   | enabled |                 connstr                  | max_sample_age | last_sample_id 
-----------+----------------+----------------------+---------------+---------+------------------------------------------+----------------+----------------
         2 | srvtestrenamed | Server description 2 | {db1,db2,db3} | f       | dbname=postgres host=localhost port=5433 |              3 |              0
(1 row)

SELECT profile.enable_server('srvtestrenamed');
 enable_server 
---------------
             1
(1 row)

SELECT server_id, server_name, server_description, db_exclude,
  enabled, connstr, max_sample_age, last_sample_id
FROM profile.servers WHERE server_name != 'local';
 server_id |  server_name   |  server_description  |  db_exclude   | enabled |                 connstr                  | max_sample_age | last_sample_id 
-----------+----------------+----------------------+---------------+---------+------------------------------------------+----------------+----------------
         2 | srvtestrenamed | Server description 2 | {db1,db2,db3} | t       | dbname=postgres host=localhost port=5433 |              3 |              0
(1 row)

SELECT * FROM profile.show_servers() where server_name != 'local';
  server_name   |                 connstr                  | enabled | max_sample_age |     description      
----------------+------------------------------------------+---------+----------------+----------------------
 srvtestrenamed | dbname=postgres host=localhost port=5433 | t       |              3 | Server description 2
(1 row)

SELECT * FROM profile.drop_server('srvtestrenamed');
 drop_server 
-------------
           1
(1 row)

